numba/core/options.py

```diff
@@ -43,16 +43,23 @@ def set_flags(self, flags):
         if kws.pop('looplift', True):
             flags.set("enable_looplift")
 
-        if kws.pop('boundscheck', False):
-            flags.set("boundscheck")
-
         if kws.pop('_nrt', True):
             flags.set("nrt")
 
-        if kws.pop('debug', config.DEBUGINFO_DEFAULT):
+        debug_mode = kws.pop('debug', config.DEBUGINFO_DEFAULT)
+        bounds_check_explicit = 'boundscheck' in kws
```

Comment:
`boundscheck` is always set in https://github.com/numba/numba/blob/0ac11479142ecbf74bbf2ca99d89be9e1d9a1f36/numba/core/decorators.py#L26-L27

Reply:
consider these tests:

https://gist.github.com/sklam/23303f2413266e2e5238cacaf5669f99